main.c

```diff
@@ -81,7 +90,22 @@ unsigned char write_buffer[4096];
 
 void apply_pin_state(const pio_spi_inst_t *spi, bool state) {
     pio_spi_enable_outputs(spi->pio, spi->sm, state, PIN_SCK, PIN_MOSI, PIN_MISO);
-    gpio_set_dir(PIN_CS, state? GPIO_OUT : GPIO_IN);
+    gpio_set_dir(cs_pins[active_cs_pin], state? GPIO_OUT : GPIO_IN);
```

Comment:
This raises an interesting design question: Should the outputs for alternative CS pins all be enabled, or just the one that is currently active used?

- If all outputs are enabled:
  - non-active CS pins will explicitly be deasserted, rather than being left floating
  - this should not cause a conflict with other SPI masters on the bus (i.e. a CPU/FPGA on a devboard), because other SPI masters must be turned off anyway in order to avoid interference on the SPI bus
- If only the selected output is enabled:
  - non-active CS pins might fluctuate if not pulled up sufficiently, and the chips connected to them might be enabled, which might cause interference on the bus.

---

If `apply_pin_state` enables/disables all pins, the code for these two functions becomes fairly simple:

```c
void apply_pin_state(const pio_spi_inst_t *spi, bool state) {
    pio_spi_enable_outputs(spi->pio, spi->sm, state, PIN_SCK, PIN_MOSI, PIN_MISO);
    for (uint i = 0; i < NUM_CS_AVAILABLE; i++) {
        gpio_set_dir(cs_pins[i], state? GPIO_OUT : GPIO_IN);
    }
    current_pin_state = state;
}

static void set_cs_pin(uint8_t cs) {
    active_cs_pin = cs;
}
```

... which makes `set_cs_pin` short enough to inline.